fuel/datasets/hdf5.py

```diff
@@ -411,21 +616,27 @@ def _in_memory_get_data(self, state=None, request=None):
         return data, shapes
 
     def _out_of_memory_get_data(self, state=None, request=None):
+        if not isinstance(request, (slice, list)):
```

Comment:
I think you're right. I remember @bartvm talking about this in another PR. If it's true, then I'll remove support for slices.